nmrfx-analyst-gui/src/main/java/org/nmrfx/analyst/gui/regions/RegionsTable.java

```diff
@@ -0,0 +1,234 @@
+            formatString = "%." + decimalPlaces + "f";
+        }
+
+        @Override
+        public String toString(Double object) {
+            return String.format(formatString, object);
+        }
+
+        @Override
+        public Double fromString(String string) {
+            try {
+                return Double.parseDouble(string);
+            } catch (NumberFormatException e) {
+                return null;
+            }
+        }
+    }
+
+    public RegionsTable() {
+        setPlaceholder(new Label("No regions to display"));
+
+        this.datasetRegions = FXCollections.observableList(new ArrayList<>());
+        SortedList<DatasetRegion> sortedRegions = new SortedList<>(this.datasetRegions);
+        sortedRegions.comparatorProperty().bind(comparatorProperty());
+        regionListener  = updateRegion -> {
+            datasetRegions.sort(Comparator.comparing(dr -> dr.getRegionStart(0)));
+            refresh();
+        };
+
+        setEditable(true);
+        TableColumn<DatasetRegion, String>  regionsLabelCol = new TableColumn<>("Region");
+        regionsLabelCol.setCellValueFactory(cellData -> new SimpleObjectProperty<>("Region " + (datasetRegions.indexOf(cellData.getValue()) + 1)));
+        getColumns().add(regionsLabelCol);
+
+
+        TableColumn<DatasetRegion, Double> startPosCol = new TableColumn<>(REGION_START_COLUMN_NAME);
+        startPosCol.setCellValueFactory(cellData -> new SimpleObjectProperty<>(cellData.getValue().getRegionStart(0)));
+        startPosCol.setCellFactory(TextFieldTableCell.forTableColumn(new DoubleColumnFormatter(NUMBER_DECIMAL_PLACES_REGION_BOUNDS)));
+        startPosCol.setEditable(true);
+        startPosCol.setOnEditCommit(this::regionBoundChanged);
+        getColumns().add(startPosCol);
+
+        TableColumn<DatasetRegion, Double> endPosCol = new TableColumn<>(REGION_END_COLUMN_NAME);
+        endPosCol.setCellValueFactory(cellData -> new SimpleObjectProperty<>(cellData.getValue().getRegionEnd(0)));
+        endPosCol.setCellFactory(TextFieldTableCell.forTableColumn(new DoubleColumnFormatter(NUMBER_DECIMAL_PLACES_REGION_BOUNDS)));
+        endPosCol.setEditable(true);
+        endPosCol.setOnEditCommit(this::regionBoundChanged);
+        getColumns().add(endPosCol);
+
+        TableColumn<DatasetRegion, Double> integralCol = new TableColumn<>(INTEGRAL_COLUMN_NAME);
+        integralCol.setCellValueFactory(new PropertyValueFactory<>("integral"));
+        integralCol.setCellFactory(column -> new DoubleTableCell(NUMBER_DECIMAL_PLACES_INTEGRAL));
+        getColumns().add(integralCol);
+
+        TableColumn<DatasetRegion, Double> normalizedIntegralCol = new TableColumn<>(NORMALIZED_INTEGRAL_COLUMN_NAME);
+        normalizedIntegralCol.setCellValueFactory(param -> {
+            double norm = chart.getDataset().getNorm() / chart.getDataset().getScale();
+            Double normProp = norm != 0.0 ? param.getValue().getIntegral() / norm : null;
```

Comment:
sonar flagged this as a bug, is it alright to use != on doubles when trying to avoid a divide by zero? Or should I check with precison...and to what precision should I check?

Reply:
Something like Math.abs(norm) > 1.0e-9 should silence sonar, and be reasonable